boards/shields/x_nucleo_iks01a2/x_nucleo_iks01a2.overlay

```diff
@@ -27,7 +27,7 @@
 	};
 
 	lsm303agr-magn@1e {
-		compatible = "st,lis2mdl","st,lsm303agr-magn";
+		compatible = "st,lsm303agr-magn", "st,lis2mdl";
```

Comment:
What is the reason to invert the devices in the compatible list?

Reply:
Because standard practice is to order them from most-specific to least-specific, selecting the first one that is supported.  There is no `lsm303agr-magn` driver right now, but there is a `lis2mdl` that works with that device.

Historically Zephyr only paid attention to the first entry, but that was fixed some months ago so we can now be consistent with other devicetree usage.

`binding-template` specifies the order, but does not explain why it's that way.

Reply:
@avisconti please revisit this; note @galak's non-verbal support for this change.